src/routes/(infoPages)/[topTierPage]/[...serviceGroupPage]/+page.server.ts

```diff
@@ -101,64 +131,122 @@ const query = gql`
-      (serviceGroup) => {
-        return serviceGroup?.pageMetadata?.slug === slug;
-      }
-    );
-    if (matchedServiceGroupsFromSlug && matchedServiceGroupsFromSlug.length > 0) {
-      let matchedPageMetadata;
-      // account for possibility that two service groups have the same ending slug
-      const matchedServiceGroup = matchedServiceGroupsFromSlug.find((group) => {
-        const serviceGroupMetadataId = group?.pageMetadata?.sys?.id || "";
-        matchedPageMetadata = pageMetadataMap.get(serviceGroupMetadataId);
-        if (matchedPageMetadata) {
-          return matchedPageMetadata.url === url;
-        }
-      });
+type ServiceGroup = ExtractQueryType<
+  ServiceGroupCollectionQuery,
+  ["serviceGroupCollection", "items", number]
+>;
 
-      if (matchedServiceGroup) {
-        const serviceEntries = matchedServiceGroup.serviceEntriesCollection?.items.filter(
-          (item) => item?.__typename === "ServiceEntry"
-        );
-        let serviceGroups = matchedServiceGroup.serviceEntriesCollection?.items.filter(
-          (item) => item?.__typename === "ServiceGroup"
-        ) as ServiceGroup[];
+type ServiceGroupMetadata = ExtractQueryType<ServiceGroup, ["pageMetadata"]>;
 
-        serviceGroups = serviceGroups.map((serviceGroup) => {
-          const serviceGroupMetadata = pageMetadataMap.get(
-            serviceGroup?.pageMetadata?.sys.id || ""
-          );
-          return { ...serviceGroup, url: serviceGroupMetadata?.url };
-        });
+type ChildServiceEntryOrGroup = ExtractQueryType<
+  ServiceGroup,
+  ["serviceEntriesCollection", "items", number]
+>;
 
-        return {
-          ...matchedServiceGroup,
-          pageMetadata: matchedPageMetadata,
-          serviceEntries,
-          serviceGroups,
-        } as ServiceGroupPage;
-      }
-    } else {
-      console.warn(
-        `A Service Group entry with the slug "${slug}" could not be found. If this page was reached via a link, it is likely that the Page Metadata entry is published but the Service Group entry is not.`
-      );
-    }
+type ChildServiceEntry = Extract<ChildServiceEntryOrGroup, { __typename: "ServiceEntry" }>;
+
+type ChildServiceGroup = Extract<ChildServiceEntryOrGroup, { __typename: "ServiceGroup" }>;
+
+export type ServiceGroupPage = {
+  serviceGroup: ServiceGroup & {
+    heroImage?: ServiceGroup["heroImage"] & {
+      imageSource?: NonNullable<ServiceGroup["heroImage"]>["imageSource"] & {
+        blurhash?: string | null | undefined;
+      };
+    };
+    description?: ServiceGroup["description"] & {
+      blurhashes?: Record<string, string> | null | undefined;
+    };
+  };
+  childServiceEntries: (ChildServiceEntry & {
+    description?: ChildServiceEntry["description"] & {
+      blurhashes?: Record<string, string> | null | undefined;
+    };
+  })[];
+  childServiceGroups: (ChildServiceGroup & { url?: string | null | undefined })[];
+  pageMetadata?: ServiceGroupMetadata;
+};
```

Comment:
Would it make sense to export these from a `src/routes/(infoPages)/[topTierPage]/[...serviceGroupPage]/types.ts` file and import them here?